src/aggregate/functions/date.c

```diff
@@ -272,19 +272,19 @@ static int parseTime(ExprEval *ctx, RSValue *result, RSValue **argv, size_t argc
 
   char fmtbuf[1024] = {0};
   char valbuf[1024] = {0};
-  size_t fmtlen;
 
-  const char *origfmt = RSValue_StringPtrLen(argv[0], &fmtlen);
+  size_t fmtlen;
+  const char *origfmt = RSValue_StringPtrLen(argv[1], &fmtlen);
   if (fmtlen > sizeof(fmtbuf)) {
     goto err;
   }
+  memcpy(fmtbuf, origfmt, fmtlen);
```

Comment:
don't we skip the '\0` here?

Reply:
This was existing code that was just moved to make it more readable.